### PR TITLE
[Op] Add reciprocal op to RAF

### PR DESCRIPTION
**Pull request by whbldhwj**
## Description ##
This PR adds reciprocal operator to RAF and updates the document about adding an operator to RAF.

## Checklist ##

- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

cc @awslabs/raf-reviewer

### CODE REVIEW

File: python/raf/amp/type_hints.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def _gen(args, ret_type, amp_dtype):
 register_op_cast_rule("raf.op.arange", generic_cast(False, 3))
 register_op_cast_rule("raf.op.exp", generic_cast(False, 1))
 register_op_cast_rule("raf.op.power", generic_cast(False, 1))
+register_op_cast_rule("raf.op.reciprocal", generic_cast(False, 1))
```
Comment (comaniac) [inline python/raf/amp/type_hints.py:125]: Does fp16 hurt the accuracy?

File: src/op/grad/nn.cc [hunk 2]
```diff
@@ -450,6 +450,24 @@ Array<Expr> LayerNormTrainGrad(const Expr& orig_call, const Array<Expr> orig_arg
...
+  static auto op_div = Op::Get("raf.op.divide");
+  static auto op_multiply = Op::Get("raf.op.multiply");
+  static auto op_negative = Op::Get("raf.op.negative");
+  static auto op_ones = Op::Get("raf.op.ones_like");
+  const CallNode* call = orig_call.as<CallNode>();
+  CHECK_GE(call->args.size(), 1);
+  const Expr& x = call->args[0];
+  Call one = Call(op_ones, {x});
+  Call x_pow_2 = Call(op_multiply, {x, x});
+  Call one_divide_x_pow_2 = Call(op_div, {one, x_pow_2});
```
Comment (comaniac) [inline src/op/grad/nn.cc:464]: How to deal with `x=0`?

**Review by comaniac — CHANGES_REQUESTED** [review]
Will review the doc change later. 
We also need unit tests.

Also please do not change TVM commit.

File: src/op/grad/nn.cc (hunk 2 above)
Comment (whbldhwj) [reply src/op/grad/nn.cc:464]: Outputs are `inf` in this case, same behavior to PyTorch.

File: python/raf/amp/type_hints.py (hunk 1 above)
Comment (whbldhwj) [reply python/raf/amp/type_hints.py:125]: Checked https://pytorch.org/docs/stable/amp.html#autocast-op-reference
Reciprocal op will not be autocasted to fp16.

File: docs/wiki/3_dev_guide/Add-Operator.md [hunk 3]
```diff
@@ -446,73 +507,84 @@ where `make` will be invoked when building the operator, and `Execute` will be i
...
+    ],
+)
+@pytest.mark.parametrize("axis", range(-4, 4))
+@pytest.mark.parametrize(
+    "funcs",
+    [
+        [raf._op.sym.softmax, torch.softmax],
+        [raf._op.sym.log_softmax, torch.log_softmax],
+    ],
+)
```
Comment (comaniac) [inline docs/wiki/3_dev_guide/Add-Operator.md:545]: Keep the comments for parameters.

File: docs/wiki/3_dev_guide/Add-Operator.md [hunk 4]
```diff
@@ -446,73 +507,84 @@ where `make` will be invoked when building the operator, and `Execute` will be i
...
-    # Generate the dy to test the backward op (i.e., softmax_dx).
-    m_dy, t_dy = randn_torch(shape, device=device, dtype=dtype)
-    # Run backward op in PyTorch.
+    # backward
+    m_dy, t_dy = randn_torch(shape, device="cuda")
     t_y.backward(t_dy)
-    # Run backward op in RAF.
     m_y.backward(m_dy)
-    # Check their gradients.
     check(m_x.grad, t_x.grad)
```
Comment (comaniac) [inline docs/wiki/3_dev_guide/Add-Operator.md:574]: Keep the comments.

**Review by comaniac — APPROVED** [review]

**Comment by comaniac** [comment]
Thanks @whbldhwj
